telcell/celldb/pgdatabase.py

```diff
@@ -0,0 +1,414 @@
+        radio: Optional[str | Iterable[str]] = None,
+        mcc: int = None,
+        mnc: int = None,
+        count_limit: Optional[int] = 10000,
+        random_order: bool = False,
+        exclude: Optional[List[CellIdentity]] = None,
+    ) -> CellCollection:
+        """
+        Given a Point, find antennas that are in reach from this point sorted by the distance from the grid point.
+
+        :param coords: Point for which nearby antennas are retrieved
+        :param distance_limit_m: antennas should be within this range
+        :param date: select antennas that were valid at `date`
+        :param radio: antennas should be limited to this radio technology, e.g.: LTE, UMTS, GSM (accepts `str` or list
+            of `str`)
+        :param mcc: antennas should be limited to this mcc
+        :param mnc: antennas should be limited to this mnc
+        :param count_limit: maximum number of antennas to return
+        :param exclude: antenna that should be excluded from the retrieved antennas
+        :return: retrieved antennas within reach from the Point
+        """
+        qwhere = list(self._qwhere)
+        qargs = list(self._qargs)
+
+        if coords is not None and distance_limit_m is not None:
+            x, y = point_to_rd(coords)
+            qwhere.append(
+                f"ST_DWithin(rd, 'SRID=4326;POINT({x} {y})', {distance_limit_m})"
+            )
+            if distance_lower_limit_m is not None:
+                qwhere.append(
+                    f"NOT ST_DWithin(rd, 'SRID=4326;POINT({x} {y})', {distance_lower_limit_m})"
+                )
+        if date is not None:
+            qwhere.append("(date_start is NULL OR %s >= date_start)")
+            qwhere.append("(date_end is NULL OR %s < date_end)")
+            qargs.extend([date, date])
+
+        if radio is not None:
+            if isinstance(radio, str):
+                radio = [radio]
+            qwhere.append(f"({' OR '.join(['radio = %s'])})")
+            qargs.extend(radio)
+
+        if mcc is not None:
+            qwhere.append(f"mcc = {mcc}")
+        if mnc is not None:
+            qwhere.append(f"mnc = {mnc}")
+
+        if exclude is not None:
+            if isinstance(exclude, CellIdentity):
+                exclude = [exclude]
+            for addr in exclude:
+                add_qwhere, add_qargs = _build_cell_identity_query(addr)
+                qwhere.append(f"NOT ({add_qwhere})")
+                qargs.extend(add_qargs)
+
+        qorder = self._qorder
+        if random_order is not None and random_order:
+            qorder = "ORDER BY RANDOM()"
+        elif coords is not None:
+            x, y = point_to_rd(coords)
+            qorder = f"ORDER BY ST_Distance(rd, 'SRID=4326;POINT({x} {y})')"
+
+        count_limit = count_limit if count_limit is not None else self._count_limit
+
+        return PgCollection(
+            self._con, qwhere, qargs, qorder, count_limit, self._on_duplicate
+        )
+
+    def close(self):
```

Comment:
ik vind het vanaf hier nogal een ingewikkelde constructie. Zou het niet makkelijker kunnen dat we net als in eerdere repos ergens een connection aan maken en die gewoon aan de functie meegeven? with DBConnection(**config.database) as connection etc?

Reply:
Dat doen we ook, toch? Die andere arguments zijn alleen voor interne consumptie. Dit heb ik proberen duidelijker te maken in docstr